pom.xml

```diff
@@ -1570,6 +1570,29 @@
             </properties>
         </profile>
 
+        <profile>
+            <id>dist</id>
+            <properties>
+                <failOnJavaDocError>true</failOnJavaDocError>
+            </properties>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-javadoc-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>generate-javadoc</id>
+                                <goals>
+                                    <goal>javadoc-no-fork</goal>
+                                </goals>
+                                <phase>prepare-package</phase>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+
```

Comment:
I don't understand why you're doing that? I mean we don't plan to build a distribution of the legacy code?

Reply:
I don't understand the link between this change and building a distribution of the legacy code.

I'm just making sure we build the javadoc for the *new* modules when the `-Pdist` profile is enabled, which unfortunately we didn't use to.

Reply:
Ah my bad, I didn't see that it was the parent pom.